tests/core/test_kafka_consumer.py

```
@@ -2079,3 +2081,96 @@ def test_tenant_not_found(self, mock_cursor_ctx, mock_tenant_objects):
+        self.consumer.consumer = Mock()
+        self.consumer.offset_manager = Mock()
+        self.consumer.offset_manager.should_commit.return_value = False
+
+    @patch("core.kafka_consumer.relations_api_replication")
+    @patch("core.kafka_consumer.json_format.ParseDict")
+    def test_last_message_processed_time_updated_on_success(self, mock_parse_dict, mock_replication):
+        """Test that last_message_processed_time Gauge is updated after successful processing."""
+        import time
+
+        mock_response = Mock()
+        mock_response.consistency_token.token = "test-token"
```

Ruff (0.15.21)

[error] Possible hardcoded password assigned to: "token"

(S105)